FRANKDOC.md

```diff
@@ -0,0 +1,86 @@
+
+![eclipseConfigurationJava](./picturesForContributors/eclipseConfigurationJava.jpg)
+
+You see the JavaDoc comment of the class (number 1), the JavaDoc of a config child setter (number 2) and the JavaDoc of an attribute (number 3).
+
+These JavaDoc comments appear in VSCode as shown below:
+
+![vscodeConfigurationDescriptions](./picturesForContributors/vscodeConfigurationDescriptions.jpg)
+
+The JavaDoc comment of the class appears as help about XML element `<Configuration>` (number 1). The JavaDoc of the attribute setter appears as help about the attribute (number 3). The JavaDoc of the config child setter is not in the XML schema file (number 2 is not present).
+
+The JavaDoc of a config child setter is available in the web application as shown below.
+
+![webappConfigurationDescriptions](./picturesForContributors/webappConfigurationDescriptions.jpg)
+
+The JavaDoc of the class appears at the top of the page. The JavaDoc of the config child setter appears in the table of config children. And the JavaDoc of the attribute setter appears with the attributes. The name of the config child (number 4) does not come from file [Configuration.java](./core/src/main/java/nl/nn/adapterframework/configuration/Configuration.java). It comes from file [digester-rules.xml](./core/src/main/resources/digester-rules.xml), which contains the following line:
+
+![digesterRulesRoleAdapter](./picturesForContributors/digesterRulesRoleAdapter.jpg)
+
+There is a `<rule>` XML element that links the name `adapter` to the name of the config child setter, `registerAdapter`.
+
+## How Java inheritance is shown
+
+In Java, config child setters and attribute setters are inherited like any Java method. As a consequence, attributes and config children are inherited. In `FrankConfig-strict.xsd` and in the JSON, inheritance is used to avoid unnecessary repetition and reduce file sizes. Frank developers do not see this in their text editor. Attributes look the same whether they are declared or inherited, and the same is true for child elements. In the web application however, inheritance is clearly visible as shown below:
+
+![webappInheritance](./picturesForContributors/webappInheritance.jpg)
+
+Declared config children and declared attributes go first. If config children or attributes are inherited, there is a heading with the closest ancestor from which we have inheritance. The config children and the attributes inherited from that ancestor follow. Next comes a header for the next ancestor from which config children or attributes are inherited. The recursion ends when all declared and all inherited config children and attributes are present.
+
+## Preferred order of child elements
+
+Attributes can appear in any order, but there is a preferred order for config children (child elements). The previous section showed config children in their preferred order. The preferred order of config children follows from the following rules:
```

Comment:
Attributes kunnen in XML inderdaad in elke volgorde staan, maar dat lijkt me niet iets om hier zo nadrukkelijk te vermelden. We doen juist moeite om ze op een zinnige volgorde in de documentatie te vermelden.

Wat ik niet terug vind in deze manual, maar wat daarin juist wel belangrijk is, is hoe je die volgorde kan beinvloeden, door de volgorde waarin javadocs in de source staan, en/of de index parameter in de IbisDoc annotatie

Reply:
Ik heb de tekst aangepast, maar ik zou niet zonder meer "gedaan" durven schrijven. Ik hoop dat het zo goed is.